src/applications/find-forms/helpers/index.js

```diff
@@ -0,0 +1,48 @@
+import { deriveLatestIssue } from '../components/SearchResult';
+import { SORT_OPTIONS } from '../constants';
+
+/**
+ * This function sorts the results of Find Forms search.
+ * @param {string} howToSort Is the state property in the SearchResults Controller
+ * @param {object} indexA Is a Search Result returned.
+ * @param {object} indexB Is a Search Result returned.
+ */
+
+export const sortTheResults = (howToSort, indexA, indexB) => {
+  // -1 sorts indexA to the front of the array.
+  // 1 sorts indexA to the back of the array.
+  // 0 keeps both indexA and indexB right where they are.
+  const [LAST_UPDATED_NEWEST_OPTION, LAST_UPDATED_OLDEST_OPTION] = SORT_OPTIONS;
+
+  const latestTimeStampIndexA = deriveLatestIssue(
+    indexA.attributes.firstIssuedOn,
+    indexA.attributes.lastRevisionOn,
+  );
+
+  const latestTimeStampIndexB = deriveLatestIssue(
+    indexB.attributes.firstIssuedOn,
+    indexB.attributes.lastRevisionOn,
+  );
+
+  const newestDate = deriveLatestIssue(
+    latestTimeStampIndexA,
+    latestTimeStampIndexB,
+  );
+
+  const oldestDate =
+    latestTimeStampIndexA === newestDate
+      ? latestTimeStampIndexB
+      : latestTimeStampIndexA;
+
+  if (howToSort === LAST_UPDATED_NEWEST_OPTION) {
+    if (newestDate === latestTimeStampIndexA) return -1;
+    else if (newestDate === latestTimeStampIndexB) return 1;
+  }
+
+  if (howToSort === LAST_UPDATED_OLDEST_OPTION) {
+    if (oldestDate === latestTimeStampIndexA) return -1;
+    else if (oldestDate === latestTimeStampIndexB) return 1;
+  }
+
+  return 0;
```

Comment:
I don't really follow this function well but in a sort function you don't need to return those exact values - what's important is a negative/positive value or zero, so you can do date arithmetic instead of if-statements.  For example - `date1 - date2` (the dates are coerced into their millisecond value). Would that help at all here?

Reply:
So it is my understanding of the js sorting function, when doing arithmetic, it will convert the dates in to UTF-16 code units if they are not numbers. Given that case, the sort compare function would not order the date strings correctly. 

Here I write the cases out to how it should sort. 
The return numbers are just what the JS sort method requires.

Reply:
Ah sorry, I’m just not sure I follow. The dates are converted to their value in milliseconds, aren’t they? Idk what “code units” refers to? 

Little SO thread on this, https://stackoverflow.com/questions/10123953/how-to-sort-an-array-by-a-date-property#10124053

Reply:
Ah, nice find, that would work then.
It would remove the dependency on Momentjs which the sorting algo has.

Reply:
Nice find! I didn't put two and two together and just convert the moment strings to date in milliseconds.